.circleci/config.yml

```diff
@@ -238,40 +283,45 @@ jobs:
             pip3.8 install conbench
             pip3.8 install git+https://github.com/conbench/benchalerts.git@0.3.1
             pip3.8 install scripts/veloxbench
+      - run-benchmarks
+      # TODO: Disabling conbench github notifications for now until results are more stable.
+      #- run:
+      #  name: "Analyze Benchmark Regressions and Post to Github"
+      #    command: |
+      #      export GITHUB_APP_PRIVATE_KEY=$(echo $GITHUB_APP_PRIVATE_KEY_ENCODED | base64 -d)
+      #      python3.8 scripts/benchmark-github-status.py --z-score-threshold 50
+      - post-steps
+
+  linux-benchmarks-basic-dedicated:
+    machine:
+      image: ubuntu-2004:current
+    resource_class: 2xlarge
+    environment:
+      CONBENCH_URL: "https://velox-conbench.voltrondata.run/"
+      CONBENCH_HOST: "CircleCI-runner-dedicated"
+      LINUX_DISTRO: "ubuntu"
+    steps:
+      - checkout
+      - update-submodules
       - run:
-          name: "Run Benchmarks"
+          name: "Setup Ubuntu"
           command: |
-            # get to the directory with benchmarks
-            cd scripts/veloxbench/veloxbench
-
-            # setup .conbench file
-            touch .conbench
-            echo "url: $CONBENCH_URL" >> .conbench
-            echo "email: $CONBENCH_EMAIL" >> .conbench
-            echo "password: $CONBENCH_PASSWORD" >> .conbench
-            echo "host_name: CircleCI-runner" >> .conbench
-
-            # Figure out the run name and run reason
-            if [[ $CIRCLE_BRANCH == 'main' ]]; then
-              # Currently, the names of runs to be included in Conbench history must
-              # start with 'commit:'
-              # (https://github.com/conbench/conbench/issues/350)
-              # This way we can keep track of performance over time on the main branch
-              # separately from pull request branches and manual runs.
-
-              run_name="commit: $(git rev-parse HEAD)"
-              run_reason='commit'
-            else
-              run_name="pull request $CIRCLE_PR_NUMBER: $(git rev-parse HEAD)"
-              run_reason='pull request'
-            fi
-
-            conbench cpp-micro --run-name="$run_name" --run-reason="$run_reason"
+            sudo ./scripts/setup-ubuntu.sh
+      - setup-environment
+      - build-benchmarks
       - run:
-          name: "Analyze Benchmark Regressions and Post to Github"
+          name: "Install Conbench, Benchalerts, and Veloxbench"
           command: |
-            export GITHUB_APP_PRIVATE_KEY=$(echo $GITHUB_APP_PRIVATE_KEY_ENCODED | base64 -d)
-            python3.8 scripts/benchmark-github-status.py --z-score-threshold 50
+            pip3 install conbench
+            pip3 install git+https://github.com/conbench/benchalerts.git@0.3.1
+            pip3 install scripts/veloxbench
+      - run-benchmarks
+      # TODO: Disabling conbench github notifications for now until results are more stable.
```

Comment:
nit: same here.

Reply:
And for what it's worth, I suspect we won't be able to use this generally for alerting on PRs since these runs can't run on PRs (unless that restriction was lifted somehow?